File: src/main/java/org/goafabric/personservice/repository/entity/AddressEo.java
package org.goafabric.personservice.repository.entity;

import jakarta.persistence.*;
import org.goafabric.personservice.repository.extensions.AuditTrailListener;


@Entity
@Table(name="address")
@EntityListeners(AuditTrailListener.class)
public class AddressEo {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    private String street;
    private String city;

    @Version //optimistic locking
    private Long version;

    public AddressEo(String id, String street, String city, Long version) {
        this.id = id;
        this.street = street;
        this.city = city;
        this.version = version;
    }

    private AddressEo() {
    }

    public String getId() {
        return id;
    }

    public String getStreet() {
        return street;
    }

    public String getCity() {
        return city;
    }

    public Long getVersion() {
        return version;
    }
}

File: src/main/kotlin/org/goafabric/personservice/repository/entity/AddressEo.kt
package org.goafabric.personservice.repository.entity

import org.hibernate.annotations.GenericGenerator
import jakarta.persistence.*
import org.goafabric.personservice.repository.extensions.AuditTrailListener
import org.hibernate.annotations.TenantId

@Entity
@Table(name = "address")
@EntityListeners(AuditTrailListener::class)
data class AddressEo (
    @Id
    @GeneratedValue(generator = "uuid")
    @GenericGenerator(name = "uuid", strategy = "uuid2")
    val id: String? = null,

    @TenantId
    var orgunitId: String? = null,

    val street: String,
    val city: String,

    @Version //optimistic locking
    val version: Long? = null
)

